tools/syscall_header_gen.py

```diff
@@ -201,18 +193,36 @@ def convert_to_assembler_format(s):
     return '_'.join(words).upper()
 
 
+def map_syscalls_neg(syscalls):
+    # This function will map a list of (condition, syscall_list) tuples
+    # From: [(cond1, ["SyscallOne", "SysCallTwo"]),
+    #        (cond2, ["SyscallThree", ...])]
+    # Into: [(cond1, [("SyscallOne", -1), ("SyscallTwo", -2)]),
+    #        (cond2, [(SyscallThree", -3), ...])]
+    r = range(1, sys.maxsize)
```

Comment:
Python has `itertools` that makes this easier. With array comprehensions there is no need to use map() and zip() either 
```
import itertools
....   

def map_syscalls_neg(syscalls):
    r = itertools.count(start = -1, step = -1)
    return [ (cond, [ (s, next(r)) for s in lst ]) for (cond, lst) in syscalls  ] 
```

Reply:
Not much of a python person so thanks for these, they seem a bit cleaner.